lib/proxy/helpers.rb

```diff
@@ -7,16 +7,18 @@ module Proxy::Helpers
   # Accepts a html error code and a message, which is then returned to the caller after adding to the proxy log
   # OR  a block which is executed and its errors handled in a similar way.
   # If no code is supplied when the block is declared then the html error used is 400.
-  def log_halt code=nil, exception=nil
-    message = exception.is_a?(String) ? exception : exception.to_s
+  def log_halt code=nil, exception_or_msg=nil, custom_msg=nil
+    message = exception_or_msg.is_a?(String) ? exception_or_msg : exception_or_msg.to_s
+    message = "#{custom_msg}: #{message}" if custom_msg
+    exception = exception_or_msg.is_a?(Exception) ? exception_or_msg.message : exception_or_msg
     begin
       if block_given?
         return yield
       end
     rescue => e
       exception = e
       message += e.message
-      code     = code || 400
+      code = code || 400
```

Comment:
Sort of related is that I'd deprecate the calling of `log_halt` without a code. Probably better to do it in a separate PR, but it looks like everything in the proxy uses an explicit code anyway (as you should).

Reply:
I am fine doing it in this PR however this will break all plugins, almost all of them use this and can possibly use this. Deprecating could do it but I feel like it is not probably worth it.

Reply:
I also would like to see that deprecation. A quick grep though the plugins I have checked out all use log_halt with a code so in 1.25 we could make it a required parameter.